include/fmt/format.h

```diff
@@ -3674,21 +3674,39 @@ struct arg_join : detail::view {
       : begin(b), end(e), sep(s) {}
 };
 
+template <typename Char>
+using iterator_arg_mapper = detail::arg_mapper<FMT_BUFFER_CONTEXT(Char)>;
+
+template <typename It, typename Char>
+using iterator_format_type =
+    decltype(std::declval<iterator_arg_mapper<Char>>().map(
+        std::declval<typename std::iterator_traits<It>::value_type>()));
+
+template <typename It, typename Char>
+using iterator_formatter = conditional_t<
```

Comment:
This is definitely not so nice.
Also TBH I'm not sure that using `FMT_BUFFER_CONTEXT` is correct here.